modules/b2b_logic/logic.c

```diff
@@ -476,23 +476,43 @@ void b2b_mark_todel( b2bl_tuple_t* tuple)
 
 int b2b_get_local_contact(struct sip_msg *msg, str *from_uri, str *local_contact)
 {
-	struct sip_uri ct_uri;
+	struct sip_uri ct_uri,server_uri;
 	const struct socket_info *send_sock = msg ?
 		(msg->force_send_socket?msg->force_send_socket:msg->rcv.bind_address):NULL;
 
 	if (server_address.len) {
 		if (pv_printf_s(msg, server_address_pve, local_contact) != 0) {
 			LM_WARN("Failed to print format string from 'server_address'\n");
-
-			if (msg) {
-				get_local_contact(send_sock, NULL, local_contact);
-			} else {
-				LM_ERR("No current SIP message, "
-					"failed to build Contact from send socket\n");
-				return -1;
+			goto msg_contact;
+		} else {
+			/* validate what we have built */
+			if (parse_uri(local_contact->s, local_contact->len, &server_uri) < 0) {
```

Comment:
parsing the generated URI (based on server_address_pve) is an unnecessary burn - as script writer you need to be sure you configuration is correct and not to expect OpenSIPS to parse it (for each message) just to be sure you are right - eventually you can do a dummy test at startup, to see if the pve generates a valid URI